subsys/net/ip/tcp2.c

```diff
@@ -1260,6 +1328,150 @@ static bool tcp_validate_seq(struct tcp *conn, struct tcphdr *hdr)
 		(net_tcp_seq_cmp(th_seq(hdr), conn->ack + conn->recv_win) < 0);
 }
 
+static void print_seq_list(struct net_buf *buf)
+{
+	struct net_buf *tmp = buf;
+	uint32_t seq;
+
+	while (tmp) {
+		seq = tcp_get_seq(tmp);
+
+		NET_DBG("buf %p seq %u len %d", tmp, seq, tmp->len);
+
+		tmp = tmp->frags;
+	}
+}
+
+static struct net_buf *tcp_queue_recv_data(struct tcp *conn,
+					   struct net_pkt *pkt,
+					   size_t len,
+					   uint32_t seq)
+{
+	struct net_buf *buf = NULL;
```

Comment:
I might be missing something, but I don't see where `buf` would be set to something other than NULL. This makes the logic at line 1457 never executed.

Reply:
Thanks, you just found a bug :)

Reply:
I made this function void as it does not need to return anything.